QuizApp/FinishedQuiz/Presenter/QuizFinishedPresenter.swift

```diff
@@ -0,0 +1,56 @@
+//
+//  QuizFinishedPresenter.swift
+//  QuizApp
+//
+//  Created by Luka Cicak on 15.05.2021..
+//
+
+import Foundation
+import UIKit
+
+class QuizFinishedPresenter {
+    
+    func showAlert(viewCont: QuizFinishedViewController) {
+        let alertController = UIAlertController(title: "Congrats", message:
+            "Your score has been uploaded.", preferredStyle: .alert)
+        alertController.addAction(UIAlertAction(title: "Dismiss", style: .default))
+        
+        DispatchQueue.main.async {
+            viewCont.present(alertController, animated: true, completion: nil)
+        }
+    }
+    
+    func sendResults(viewCont: QuizFinishedViewController){
+    
+        DispatchQueue.main.async {
+            let url = URL(string: "https://iosquiz.herokuapp.com/api/result")!
+            var request = URLRequest(url: url)
+            let networkService = NetworkService()
+            let json: [String: Any] = ["quiz_id":UserDefaults.standard.integer(forKey: "quiz_id"),
+                                       "user_id": UserDefaults.standard.integer(forKey: "user_id"),
+                                       "time": UserDefaults.standard.double(forKey: "time"),
+                                       "no_of_correct": UserDefaults.standard.integer(forKey: "no_of_correct")]
+            
+            let jsonData = try? JSONSerialization.data(withJSONObject: json)
+            
+            request.httpBody = jsonData
+            request.httpMethod = "POST"
+            request.setValue(UserDefaults.standard.string(forKey: "token"), forHTTPHeaderField: "Authorization")
+            request.setValue("application/json", forHTTPHeaderField: "Content-Type")
+            
+        
+            networkService.executeUrlRequest(request) { (result: Result<ResultResponse, RequestError>) in
```

Comment:
This is the first place I've noticed this, but please check the whole code.
`executeUrlRequest` has an escaping closure for its parameter. That means that that closure will outlive the function and create a memory leak if you're using `strong self` here (which you are on line 49).
What will  happen is following:
the closure that's referencing `self` will increase its reference count and if that closure doesn't get deallocated before the `QuizFinishedPresenter` it will keep it in the memory while you think that it's gone.

This is explained in the materials you got  for Memory Management in Swift.
To remedy this you can do the following:
```swift
networkService
    . executeUrlRequest(request) { [weak self] result in 
        guard let self = self else { return }
    
        // the rest of the code
    }
```